pkg/quarterdeck/api/v1/api.go

```diff
@@ -114,15 +116,19 @@ type APIAuthentication struct {
 //===========================================================================
 
 type APIKey struct {
-	ID           int      `json:"id,omitempty"`
-	ClientID     string   `json:"client_id"`
-	ClientSecret string   `json:"client_secret,omitempty"`
-	Name         string   `json:"name"`
-	ProjectID    string   `json:"project_id"`
-	Owner        string   `json:"owner,omitempty"`
-	Permissions  []string `json:"permissions,omitempty"`
-	Created      string   `json:"created,omitempty"`
-	Modified     string   `json:"modified,omitempty"`
+	ID           ulid.ULID `json:"id,omitempty"`            // not allowed on create
+	ClientID     string    `json:"client_id"`               // not allowed on create, cannot be updated
+	ClientSecret string    `json:"client_secret,omitempty"` // not allowed on created, cannot be updated
+	Name         string    `json:"name"`                    // required on create, update
+	OrgID        ulid.ULID `json:"org_id"`                  // required on create, cannot be updated
+	ProjectID    ulid.ULID `json:"project_id"`              // required on create, cannot be updated
+	CreatedBy    ulid.ULID `json:"owner,omitempty"`         // required on create, cannot be updated
```

Comment:
Is the `CreatedBy` datatype correct? It's listed as `sql.NullByte` in the database model instead of `ulid.ULID`

Reply:
`sql.NullByte` is the only way we can have a nullable ULID -- the database storage for a ULID is `BLOB` (bytes) - the `ULID` is a `[16]byte` so it automatically gets converted to `[]byte` when going into and out of the database and a type conversion happens when it is retrieved. If we used `ulid.ULID` for the nullable field, then we would always insert `0x00 x16` into the database. 

I guess the question is if an API Key can be created without an owner account -- if not, we could always make it NOT NULL.